compiler/vm/compilerbridge.nim

```diff
@@ -274,12 +353,23 @@ template returnOnErr(res: VmGenResult, config: ConfigRef, node: PNode): CodeInfo
   if r.isOk:
     r.take
   else:
-    let
-      report = vmGenDiagToLegacyVmReport(r.takeErr)
-      kind = report.kind
-      rid = config.addReport(wrap(report))
+    # let
+    #   report = vmGenDiagToLegacyVmReport(r.takeErr)
+    #   kind = report.kind
+    #   rid = config.addReport(wrap(report))
 
-    return config.newError(node, kind, rid, instLoc())
+    # return config.newError(node, kind, rid, instLoc())
+    let
+      vmGenDiag = r.takeErr
+      diag = PAstDiag(
+              kind: adVmGenError,
+              location: vmGenDiag.location,
+              instLoc: vmGenDiag.instLoc,
+              vmGenErr: vmGenDiagToAstDiagVmGenError(vmGenDiag),
+              duringJit: false)
+
+    # QUESTION: Why are we overriding the instantiation location again?
+    return config.newError(node, diag, instLoc())
```

Comment:
Actually realizing now, I'm probably the one who overrode this `instLoc` in the first place. 😆